lib/rototiller/utilities/env_var.rb

```diff
@@ -0,0 +1,55 @@
+require 'rototiller/utilities/color_text'
+
+class EnvVar
+
+  include ColorText
+
+  attr_accessor :var, :message, :default
+
+  def initialize(var, message, default = nil)
+    # require 'pry'; binding.pry
+    @var = var
+    @message = message
+    # TODO: maybe see if the value was passed in the args array or not as a default (so that nil may be passed)
+    # should be something like argv or similar
+    # @default_specified = ????
+    @default = default
+    #test = ENV[var]
+    #binding.pry
+    #if test
+    #  @value = test
+    #else
+    #  @value = @default
+    #end
+    #binding.pry
+  end
+
+  def value
+    ENV[@var] || @default
+  end
+
+  def message
+
+    if @default && check
+      # ENV is present and it has a default value
+      # binding.pry
+      green_text("The ENV #{@var} is being used with the value #{value} from the environment")
+    elsif !@default && !check
+      # ENV is not Present and it has no default value
+      red_text("The ENV #{@var} is required, #{@message}")
+    elsif !@default && check
+      # ENV is present and it has no default value
+      green_text("The ENV #{@var} was found in the environment with the value #{value}")
+    elsif @default && !check
+      # ENV is not present and it has default value
+      yellow = yellow_text("WARNING: the ENV #{@var} is not set, proceeding with default value: ")
+      green = green_text(value)
+      yellow << green
+    end
+  end
+
+  private
+  def check
```

Comment:
should this set a status instance var, and should it check if there is default?
maybe check is just for existence of ENV and something like is_valid?() (public) would tell if the var is usable?  then the collector would know if to spew the warnings/errors.
i'd be okay doing that when we implement the collector